tools/dotnet-linker/BackingFieldDelayHandler.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using Xamarin.Tuner;
+
+namespace Xamarin.Linker {
+
+	// Problems:
+	// * `Dispose` set the generated backing fields to `null` which means
+	//   the linker will mark every backing fields, even if not used 
+	//   elsewhere (generally properties) inside the class
+	// * Backing fields increase the memory footprint of the managed peer
+	//   instance (for the type and all it's subclasses)
+	// * Backing fields also increase the app size. Not a huge problem as
+	//   they are all declared _weakly_ as `NSObject` but still...
+	//
+	// Solution:
+	// * When the linker process a `Dispose` method of an `NSObject` 
+	//   subclass with the _optimizable_ attribute then we remove the
+	//   method body. This way the linker cannot mark the fields.
+	// * Before saving back the assemblies we replace the cached method
+	//   body and NOP every field that were not marked by something else
+	//   than the `Dispose` method.
+	public class BackingFieldDelayHandler : ConfigurationAwareMarkHandler {
+
+		protected override string Name { get; } = "Backing Fields Optimizer";
+		protected override int ErrorCode { get; } = 2200;
```

Comment:
2200 is taken: https://github.com/xamarin/xamarin-macios/blob/d11852e02c3d51be71b6989e0eacb07e945b0e82/tools/linker/MonoTouch.Tuner/PreserveSmartEnumConversions.cs#L28

Reply:
oops, forgot to update it